src/system/SystemObject.h

```diff
@@ -212,8 +303,25 @@ inline T * ObjectPool<T, N>::Get(const Layer & aLayer, size_t aIndex)
 {
     T * lReturn = nullptr;
 
+#if CHIP_SYSTEM_CONFIG_POOL_USE_HEAP
```

Comment:
Can we just remove the indexed getter altogether?  It doesn't make sense for the heap implementation, so is impossible to include in a single uniform interface.  And it appears to be no longer necessary with the `ForEachActiveObject` method.

If not a wholesale removal, can we at least remove it from the `CHIP_SYSTEM_CONFIG_POOL_USE_HEAP` build?  There's no way to sensibly use this in that build.  All it gives you is a random object that happens to be at the given index at that moment.  It's impossible to write code that would do anything repeatable with this.

Reply:
I have removed the usage of indexed getter usage throughout the stack except unit test,  I still keep this interface since the whole TestSystemObject is written against this, I plan to rewrite the unit test (quite complicated) in the follow-up PR and will remove indexed getter interface there.

Reply:
OK, that's fair.  Thank you.